lib/std/math/log_int.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("../std.zig");
+const math = std.math;
+const testing = std.testing;
+const assert = std.debug.assert;
+const Log2Int = math.Log2Int;
+
+/// Returns the logarithm of `x` for the provided `base`, rounding down to the nearest integer.
+/// Asserts that `base > 1` and `x != 0`.
+pub fn log_int(comptime T: type, base: T, x: T) Log2Int(T) {
+    if (@typeInfo(T) != .Int or @typeInfo(T).Int.signedness != .unsigned)
+        @compileError("log_int requires an unsigned integer, found " ++ @typeName(T));
+
+    assert(base > 1 and x != 0);
+
+    var exponent: Log2Int(T) = 0;
+    var power: T = 1;
+    while (power <= x / base) {
+        power *= base;
+        exponent += 1;
+    }
+
+    return exponent;
+}
+
+test "log" {
+    // test all unsigned integers with 2, 3, ..., 64 bits
+    inline for (2..64 + 1) |bits| {
+        const T = @Type(std.builtin.Type{
+            .Int = std.builtin.Type.Int{ .signedness = .unsigned, .bits = @intCast(bits) },
+        });
+
+        // for base = 2, 3, ..., min(maxInt(T),1024)
+        var base: T = 1;
+        while (base < math.maxInt(T) and base <= 1024) {
+            base += 1;
+
+            // test `log(1) == 0`
+            try testing.expectEqual(@as(Log2Int(T), 0), log_int(T, base, 1));
+
+            // for powers `pow = base^exp` that fit inside T
+            var exp: Log2Int(T) = 0;
+            var pow: T = 1;
+            while (pow < math.maxInt(T) / base) {
+                exp += 1;
+                pow *= base;
+
+                // test that `log_int` correctly detects the threshold
```

Comment:
what threshold? I think this is simpler to understand, if you combine it with above comment (for power ..).

Reply:
Yeah I agree :)